plugins/draco/module/vtkF3DDracoReader.cxx

```diff
@@ -0,0 +1,207 @@
+          break;
+        case draco::DT_INT64:
+          dataArray = vtkInternals::FillArray<long long>(nbPoints, attr);
+          break;
+        case draco::DT_UINT64:
+          dataArray = vtkInternals::FillArray<unsigned long long>(nbPoints, attr);
+          break;
+        case draco::DT_FLOAT32:
+          dataArray = vtkInternals::FillArray<float>(nbPoints, attr);
+          break;
+          break;
+        case draco::DT_FLOAT64:
+          dataArray = vtkInternals::FillArray<double>(nbPoints, attr);
+          break;
+        default:
+          break;
+      }
+
+      if (dataArray)
+      {
+        switch (attr->attribute_type())
+        {
+          case draco::GeometryAttribute::Type::POSITION:
+          {
+            vtkNew<vtkPoints> points;
+            points->SetData(dataArray);
+            output->SetPoints(points);
+          }
+          break;
+          case draco::GeometryAttribute::Type::NORMAL:
+            output->GetPointData()->SetNormals(dataArray);
+            break;
+          case draco::GeometryAttribute::Type::COLOR:
+            output->GetPointData()->SetScalars(dataArray);
+            break;
+          case draco::GeometryAttribute::Type::TEX_COORD:
+            output->GetPointData()->SetTCoords(dataArray);
+            break;
+          default:
+            output->GetPointData()->AddArray(dataArray);
+            break;
+        }
+      }
+    }
+  }
+
+  static void FillFaces(const std::unique_ptr<draco::Mesh>& mesh, vtkPolyData* output)
+  {
+    int nbCells = mesh->num_faces();
+
+    vtkNew<vtkIdTypeArray> triangles;
+    triangles->SetNumberOfTuples(4 * nbCells);
+
+    for (int i = 0; i < nbCells; i++)
+    {
+      const draco::Mesh::Face& face = mesh->face(draco::FaceIndex(i));
+
+      triangles->SetTypedComponent(4 * i, 0, 3);
+      for (int j = 0; j < 3; j++)
+      {
+        triangles->SetTypedComponent(4 * i + j + 1, 0, face[j].value());
+      }
+    }
+
+    vtkNew<vtkCellArray> cells;
+    cells->SetCells(nbCells, triangles);
+
+    output->SetPolys(cells);
+  }
+
+  vtkF3DDracoReader* Parent;
+};
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro(vtkF3DDracoReader);
+
+//----------------------------------------------------------------------------
+vtkF3DDracoReader::vtkF3DDracoReader()
+  : Internals(new vtkF3DDracoReader::vtkInternals(this))
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//----------------------------------------------------------------------------
+vtkF3DDracoReader::~vtkF3DDracoReader() = default;
+
+//----------------------------------------------------------------------------
+int vtkF3DDracoReader::RequestData(
+  vtkInformation*, vtkInformationVector**, vtkInformationVector* outputVector)
+{
+  vtkPolyData* output = vtkPolyData::GetData(outputVector);
+
+  std::vector<char> data;
+
+  auto reader = draco::StdioFileReader::Open(this->FileName);
+  if (!reader->ReadFileToBuffer(&data))
```

Comment:
can this throw ?

Reply:
generic question about draco

Reply:
It's not documented as such. It returns nullptr in case of failure.